pm_muairss_write/pm_muairss_write.xml

```diff
@@ -64,10 +70,19 @@
             </when>
             <when value="false"/>
         </conditional>
+        <param type="select" name="allpos_ext" label="All Positions File" help="Optionally generate output file containing all muon sites using the specified file type." optional="true">
+            <option value="cell">CELL</option>
+            <option value="cif">CIF</option>
+            <option value="xyz">XYZ (standard)</option>
+            <option value="extxyz">XYZ (extendend)</option>
+        </param>
```

Comment:
I think it might be a good idea to hide this select in a conditional - like `iscastep` 

```suggestion
       <conditional name="isallpos">
            <param type="boolean" value="false" name="allpos_sel" label="Generate output file containing all muon sites?" help="Select 'Yes' if you want to generate additional output file containing all muon sites "/>
            <when value="true">
                <param type="select" name="allpos_ext" label="All positions file type" help="Select file type for file containing all muon sites">
                    <option value="cell">CELL</option>
                    <option value="cif">CIF</option>
                    <option value="xyz">XYZ (standard)</option>
                    <option value="extxyz">XYZ (extendend)</option>
                </param>
            </when>
            <when value="false"/>
        </conditional>
```

Reply:
following @patrick-austin's comments, I'm also not sure about adding the conditional – it seems like extra clicks and more complex code for not much gain. I do agree with the underlying implication that this param could be more clearly optional since the Galaxy UI doesn't do much to communicate it. Suggestion below
```suggestion
        <param type="select" name="allpos_ext" label="Format for file containing all muon positions (optional)" help="If set, generates an extra output file containing all the muon positions using the specified file type." optional="true">
            <option value="none" selected="true">No format (do not write a file)</option>
            <option value="cell">CELL</option>
            <option value="cif">CIF</option>
            <option value="xyz">XYZ</option>
        </param>
```

Side note: due to the way the all_pos file is written from pymuonsuite using ASE, using `allpos.xyz` as the filename will result in an extended XYZ file - see https://gitlab.com/ase/ase/-/blob/master/ase/io/formats.py#L497. So having both XYZ options isn't necessary (I actually thought using `allpos.extxyz` might break, but it seems ok... yet the line that I think makes it ok [might actually be a bug](https://gitlab.com/ase/ase/-/blob/master/ase/io/formats.py#L954)? anyway, that's off topic)

Reply:
@bgruening Yes, I've been meaning to check if there was an open issue on the Galaxy repo around this. It'd be good to have some visible marker to distinguish between required/optional parameters, especially as this is common in many (most?) other web forms

Reply:
I haven't found anything with a cursory search, so I'll make an issue

Reply:
Thanks! I am also not aware of any issue about that.

Reply:
The `allpos_filename: allpos.$allpos_ext` part is the reason I suggested `value="xyz"` – typically, all XYZ files use the `.xyz` extension, whether they're extended XYZ or not, so that's what ASE expects to receive (this is why `.extxyz` falls through the cracks slightly). The distinction between `xyz` and `extxyz` is something we've created at a Galaxy level – partly because of the difference in sniffers, partly because certain pymuonsuite commands won't work with standard XYZ.
All that said, since using `value="extxyz"` works ok, and it it's a bit more intuitive in terms of matching the output file datatype, I think either is fine. (realistically I don't see that part of the ASE code changing, but we can always switch later if it does)

For the selector - I foolishly didn't check the UI myself and assumed the default was blank or that it would be overridden by the default. You're right that this would be confusing and we should either make it non-optional or remove the extra option I suggested. I think either way is fine as long as there's clarity somewhere (e.g. help text) that no format selected = no file written.

Reply:
No that makes sense, I just was copying the formats from earlier in the form we list 4 of them. I suppose all of the additional detail in an `extxyz` just ends up as the "comment" if interpreted as a "standard" `xyz`, so in that sense it's always still a `.xyz` file. If `.xyz` is conventionally used I think go with that.

Reply:
@patrick-austin This conversation is basically resolved, but one of the XYZ options still needs to be removed as agreed

Reply:
@elichad Happy to just display a "standard" `xyz` as discussed. I'm less clear on the outcome in terms of the "optional" nature of this. Do we want to:
1. Wrap in a conditional that controls whether to generate the file, then containing a (non-optional) selector for type only
2. Avoid the "optional" flag and make an explicit "do not generate" option within the extension selector
3. Use the optional flag, and wait for the upstream issues to make the default "Nothing selected" clearer (either before or after merging this into dev/main)

My preference would be for either 2 or 3